core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -204,6 +213,7 @@ case object TaskResultLost extends TaskFailedReason {
 @DeveloperApi
 case object TaskKilled extends TaskFailedReason {
   override def toErrorString: String = "TaskKilled (killed intentionally)"
+  override def countTowardsTaskFailures: Boolean = false
```

Comment:
the default in `TaskFailedReason` is true, so it will be true for `ExceptionFailure`, `UnknownReason`, `ExecutorLost` if caused by app (eg., an OOM), `TaskResultLost`, and `Resubmitted`.  I admit I'm not certain if those last two should count or not, but changing that behavior should probably be separate from this change.  I'm not changing the end behavior here, just trying to refactor it to make it easier to follow and handle correctly with the blacklist (it was special-cased before: https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L787)

Or do you mean we should change it to a `val` instead, since its a constant?  then I'd change `TaskCommitDenied` to use a `val` as well.